### ISSUE TITLE
cdc/roachtest: add performance tests for changefeeds

### ISSUE THREAD

**Issue by jayshrivastava** [cockroachdb/cockroach#89977]
See https://docs.google.com/document/d/1BE5wbSmylb5Fb-eLxbH3Pkht0w4Mr78aFIsw4llSK-E/edit

**Comment by jayshrivastava** [comment]
The scope of this issue is to have a way to a) easily check for perf regressions, and b) have benchmarks/metrics ready to show customers/sales when they have questions. Option 3 below is the best way to do this.

We have existing roachtests (ex. `cdc/tpcc-1000` or even `elastic-cdc`) we can collect stats from. There's a few ways to get stats as this discussed in [this thread](https://cockroachlabs.slack.com/archives/C023S0V4YEB/p1667932428271339). 

1. Test Eng Grafana Cluster
Pros - It's persistent, collects metrics from roachtests, and puts data in the grafana UI.
Cons - Data is organized by cluster (and not roachtest name), so you have to do a lot of digging in teamcity to figure out in which teamcity cluster and timeframe to find the data. Also, roachtests are run on ephemeral clusters. I couldn't find data for past `cdc/tpcc-1000` runs bc the clusters were gone.

2. Roachtest Prometheus Dump
This involves manually finding roachtest runs in teamcity, downloading the prometheus zip from the artifacts, and launching grafana locally to view it.
Pros - A very small amount of work to implement. We just need to add a few lines to a roachtest to dump the prometheus zip and write a doc on how to open the dashboard locally.
Cons - This requires a decent amount of manual work by an engineer every week. 

3. Roachperf
Update roachtests to generate the `stats.json` artifact. Update roachperf to collect this stats file from our roachtests.
Pros: Roachperf is a persistent dashboard anyone can see to check for perf regressions. This dashboard can also be used to shared easily with other teams/sales.
Cons: Takes a little bit of work to implement for the first time.

**Comment by miretskiy** [comment]
3 sounds good?

**Comment by jayshrivastava** [comment]
@miretskiy Yes. 3 is the best option. I have some WIP [in the crdb repo](https://github.com/jayshrivastava/cockroach/commit/d716fefbff66435dde7c0223710feffd9f48ee7b) and [in the roachperf repo](https://github.com/cockroachdb/roachperf/commit/18f01fbea0fcf611e9ae95f55fc7858f71c236b2). These changes are based on Shiraka's work. I'll sync with him before I PR them. I'll demo the graphs today.
